File: binaryTree.cpp
#include<stdio.h>
#include<algorithm>
#include<stdlib.h>
using namespace std;

struct node
{
	int data;
	struct node *left;
	struct node *right;
};

struct node *newnode(int data)
{
	struct node* node = (struct node *)malloc(sizeof(struct node));
	node->data = data;
	node->left = NULL;
	node->right = NULL;
	return(node);
}

void preOrder(struct node *node)
{
	if(node != NULL)
	{	
		printf("%d ",node->data);
		preOrder(node->left);
		preOrder(node->right);
	}
}

void postOrder(struct node *node)
{
	if(node != NULL)
	{
		postOrder(node->left);
		postOrder(node->right);
		printf("%d ",node->data);
	}
}

void inOrder(struct node *node)
{
	if(node != NULL)
	{
		inOrder(node->left);
		printf("%d ",node->data);
		inOrder(node->right);
	}
}

int main()
{
	struct node *root = newnode(1);
	root->left = newnode(2);
	root->right = newnode(3);
	root->left->left = newnode(4);
 	root->left->right = newnode(5); 

	printf("\n Preorder traversal of binary tree is \n");
        preOrder(root);
 
        printf("\n Inorder traversal of binary tree is \n");
        inOrder(root);  
 
        printf("\n Postorder traversal of binary tree is \n");
        postOrder(root);
 
	return 0;
}

File: BinaryTree.java
import java.util.Scanner;
class node
{
	int data;
	node left;
	node right;
		
	public node(int data)
	{
		this.data = data;
		this.left = null;
		this.right = null;
	}

	public void setData(int data)
	{
		this.data = data;
	}
		
	public int getData()
	{
		return data;
	}

	public void setLeft(node n)
	{
		this.left = n;
	}

	public node getLeft()
	{
		return left;
	}

	public void setRight(node n)
	{
		this.right = n;
	}

	public node getRight()
	{
		return right;
	}
}

class tree
{
	node root;
	
	public tree()
	{
		this.root = null;
	}

	public void insert()
	{
		node x = new node(4);
		node y = new node(5);
		node z = new node(6);
                x.setLeft(y);
                x.setRight(z);
		y = new node(7);
		z = new node(8);
                x.getLeft().setLeft(y);
                x.getLeft().setRight(z);
		y = new node(9);
		z = new node(10);
                x.getRight().setLeft(y);
                x.getRight().setRight(z);

		inOrder(x);
		preOrder(x);
		postOrder(x);
	}

	public void inOrder(node n)
	{
		if(n != null)
		{
			inOrder(n.getLeft());
			System.out.println(n.getData());
			inOrder(n.getRight());
		}
	}

	public void preOrder(node n)
	{
		if(n != null)
		{
			System.out.println(n.getData());
			preOrder(n.getLeft());
			preOrder(n.getRight());
		}
	}

	public void postOrder(node n)
	{
		if(n != null)
		{
			postOrder(n.getLeft());
			postOrder(n.getRight());			
			System.out.println(n.getData());
		}		
	}
}
class main
{
	public static void main(String args[])
	{
		Scanner input = new Scanner(System.in);
		tree t = new tree();
		t.insert();
//		t.inOrder();
//		t.preOrder();
//		t.postOrder();
	}
}
